lca-cli/initmonitor/initmonitor.go

```diff
@@ -0,0 +1,107 @@
+package initmonitor
+
+import (
+	"fmt"
+	"os"
+	"time"
+
+	"github.com/sirupsen/logrus"
+	"k8s.io/apimachinery/pkg/runtime"
+
+	"github.com/openshift-kni/lifecycle-agent/internal/ostreeclient"
+	"github.com/openshift-kni/lifecycle-agent/internal/reboot"
+	"github.com/openshift-kni/lifecycle-agent/lca-cli/ops"
+	rpmostreeclient "github.com/openshift-kni/lifecycle-agent/lca-cli/ostreeclient"
+
+	"github.com/go-logr/logr"
+)
+
+type InitMonitor struct {
+	scheme               *runtime.Scheme
+	log                  *logrus.Logger
+	hostCommandsExecutor ops.Execute
+	ops                  ops.Ops
+	component            string
+	rpmOstreeClient      *rpmostreeclient.Client
+	ostreeClient         ostreeclient.IClient
+}
+
+func NewInitMonitor(scheme *runtime.Scheme, log *logrus.Logger, hostCommandsExecutor ops.Execute, ops ops.Ops, component string) *InitMonitor {
+	return &InitMonitor{
+		scheme:               scheme,
+		log:                  log,
+		hostCommandsExecutor: hostCommandsExecutor,
+		ops:                  ops,
+		component:            component,
+		rpmOstreeClient:      rpmostreeclient.NewClient("initmonitor", hostCommandsExecutor),
+		ostreeClient:         ostreeclient.NewClient(hostCommandsExecutor, false),
+	}
+}
+
+func (m *InitMonitor) RunInitMonitor() error {
+	rollbackCfg, err := reboot.ReadIBUAutoRollbackConfigFile()
+	if err != nil {
+		if os.IsNotExist(err) {
+			return nil
+		}
+
+		m.log.Error(err, "failed to read rollback config file")
+		return err
+	}
+
+	if !rollbackCfg.InitMonitorEnabled || rollbackCfg.InitMonitorTimeout <= 0 {
+		return nil
+	}
+
```

Comment:
A "Will initiate rollback if not stopped by `<timestamp>`" log could be nice here